### PR TITLE
GH-13806: add exit button to configurator in powertools

### LINKED ISSUE #13806: Create new component for "EXIT CONFIGURATION" Button in SPA Header
New component should trigger a "browser back" when clicked. 
When no browser back is possible, the user should go the the product detail of the product which is configured.

Component will be added in rulebased module, lazy loading is expected.
Component has to take care that the header is styled correctly.

### PR DESCRIPTION
**Pull request by Changsuwan**
Closes #13806

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-layout.module.ts [hunk 1]
```diff
@@ -16,10 +16,10 @@ import { CpqConfiguratorPageLayoutHandler } from './cpq-configurator-page-layout
         CpqConfigurationTemplate: {
           header: {
             md: {
-              slots: ['SiteLogo', 'MiniCart'],
+              slots: ['SiteLogo', 'VariantConfigExitButton', 'MiniCart'],
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-layout.module.ts]: Slot should be named CpqConfigExitButton

File: feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-layout.module.ts [hunk 2]
```diff
@@ -16,10 +16,10 @@ import { CpqConfiguratorPageLayoutHandler } from './cpq-configurator-page-layout
...
             },
             xs: {
-              slots: ['SiteLogo', 'MiniCart'],
+              slots: ['SiteLogo', 'VariantConfigExitButton', 'MiniCart'],
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-layout.module.ts]: CpqConfigExitButton

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-layout.module.ts (hunk 1 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-layout.module.ts]: Done

File: feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-layout.module.ts (hunk 2 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/root/cpq/cpq-configurator-layout.module.ts]: Done

**Comment by ChristophHi** [comment]
Styling is not correct in the CPQ case
Screen Shot 2021-11-04 at 16 55 44

**Review by ChristophHi — CHANGES_REQUESTED** [review]
See latest comment, styling not correct

**Review by ChristophHi — CHANGES_REQUESTED** [review]
Also: Please fix un-needed chaining operators in  product-configurator/rulebased/components/exit-button/configurator-exit-button.component.ts

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**